src/Cli/dotnet/Commands/Package/List/ListPackageReferencesCommand.cs

```diff
@@ -30,7 +32,78 @@ private static string GetAbsolutePath(string currentDirectory, string relativePa
 
     public override int Execute()
     {
-        return NuGetCommand.Run(TransformArgs());
+        string projectFile = GetProjectOrSolution();
+        bool noRestore = _parseResult.HasOption(PackageListCommandParser.NoRestore);
+        int restoreExitCode = 0;
+
+        if (!noRestore)
+        {
+            ReportOutputFormat formatOption = _parseResult.GetValue((CliOption<ReportOutputFormat>)PackageListCommandParser.FormatOption);
+            bool interactive = _parseResult.GetValue((CliOption<bool>)PackageListCommandParser.InteractiveOption);
+            restoreExitCode = RunRestore(projectFile, formatOption, interactive);
+        }
+
+        return restoreExitCode == 0
+            ? NuGetCommand.Run(TransformArgs(projectFile))
+            : restoreExitCode;
+    }
+
+    private int RunRestore(string projectOrSolution, ReportOutputFormat formatOption, bool interactive)
```

Comment:
Nit: This method does a lot of things to prepare the command to run. Consider adding private helper methods to set up the options and generate the json output and call them from this method.